services/neptune/src/test/java/software/amazon/awssdk/services/neptune/internal/PresignRequestHandlerTest.java

```diff
@@ -0,0 +1,179 @@
+            new CopyDbClusterSnapshotRequestMarshaller(RdsPresignInterceptor.PROTOCOL_FACTORY);
+
+    @Test
+    public void testSetsPresignedUrl() {
+        CopyDbClusterSnapshotRequest request = makeTestRequest();
+        SdkHttpRequest presignedRequest = modifyHttpRequest(presignInterceptor, request, marshallRequest(request));
+
+        assertNotNull(presignedRequest.rawQueryParameters().get("PreSignedUrl").get(0));
+    }
+
+    @Test
+    public void testComputesPresignedUrlCorrectlyForCopyDbClusterSnapshotRequest() {
+        // Note: test data was baselined by performing actual calls, with real
+        // credentials to RDS and checking that they succeeded. Then the
+        // request was recreated with all the same parameters but with test
+        // credentials.
+        final CopyDbClusterSnapshotRequest request = CopyDbClusterSnapshotRequest.builder()
+                .sourceDBClusterSnapshotIdentifier("arn:aws:rds:us-east-1:123456789012:snapshot:rds:test-instance-ss-2016-12-20-23-19")
+                .targetDBClusterSnapshotIdentifier("test-instance-ss-copy-2")
+                .sourceRegion("us-east-1")
+                .kmsKeyId("arn:aws:kms:us-west-2:123456789012:key/11111111-2222-3333-4444-555555555555")
+                .build();
+
+        Calendar c = new GregorianCalendar();
+        c.setTimeZone(TimeZone.getTimeZone("UTC"));
+        // 20161221T180735Z
+        // Note: month is 0-based
+        c.set(2016, Calendar.DECEMBER, 21, 18, 7, 35);
+
+        Clock signingDateOverride = Mockito.mock(Clock.class);
+        when(signingDateOverride.millis()).thenReturn(c.getTimeInMillis());
+
+        RdsPresignInterceptor<CopyDbClusterSnapshotRequest> interceptor = new CopyDbClusterSnapshotPresignInterceptor(signingDateOverride);
+
+        SdkHttpRequest presignedRequest = modifyHttpRequest(interceptor, request, marshallRequest(request));
+
+        final String expectedPreSignedUrl = "https://rds.us-east-1.amazonaws.com?" +
+                "Action=CopyDBClusterSnapshot" +
+                "&Version=2014-10-31" +
+                "&SourceDBClusterSnapshotIdentifier=arn%3Aaws%3Ards%3Aus-east-1%3A123456789012%3Asnapshot%3Ards%3Atest-instance-ss-2016-12-20-23-19" +
+                "&TargetDBClusterSnapshotIdentifier=test-instance-ss-copy-2" +
+                "&KmsKeyId=arn%3Aaws%3Akms%3Aus-west-2%3A123456789012%3Akey%2F11111111-2222-3333-4444-555555555555" +
+                "&DestinationRegion=us-west-2" +
+                "&X-Amz-Algorithm=AWS4-HMAC-SHA256" +
+                "&X-Amz-Date=20161221T180735Z" +
+                "&X-Amz-SignedHeaders=host" +
+                "&X-Amz-Expires=604800" +
+                "&X-Amz-Credential=foo%2F20161221%2Fus-east-1%2Frds%2Faws4_request" +
+                "&X-Amz-Signature=00822ebbba95e2e6ac09112aa85621fbef060a596e3e1480f9f4ac61493e9821";
+        assertEquals(expectedPreSignedUrl, presignedRequest.rawQueryParameters().get("PreSignedUrl").get(0));
```

Comment:
Same as above, can we use the `equals` method here?

Reply:
We will need to create expected presignedRequest in that case.
The other attributes are not important w.r.t test case, so we are just asserting on rawQueryParameters which is updated by interceptors